lahja/misc.py

```diff
@@ -18,6 +14,7 @@
 
 
 class Subscription:
+    __slots__ = ('_unsubscribe_fn',)
```

Comment:
TODO, extract all these `__slots__` and profile to see if they have a performance effect.

Reply:
What scares me about these slots is that they could get out of sync with the actual properties so easily. Same with typos, just feels very fragile. I wonder if something like https://github.com/cjrh/autoslot would make this cleaner?

Reply:
I don't think there's much to be scared of.  If there are extra slots then we're just taking a very small performance hit.  If there are properties not accounted for then we will get a very loud error trying to set that property.

Reply:
>If there are properties not accounted for then we will get a very loud error trying to set that property

Oh, nice! I didn't know they were all or nothing (but just tried it out to confirm). That makes them less scary indeed!